### DISCUSSION TITLE
weconnect-mqtt and  weconnect-cli installation on Android

### DISCUSSION THREAD

**Discussion by essjmcb** [tillsteinbach/WeConnect-mqtt#171] [Q&A]
Hi,
Hope this is the right place to post this question. To mention that my python knowledge is very low.

I am trying to use weconnect-cli or mqtt (cli would be enough for me), to get information from my ID4. Simple data like battery level, if it is charging or not..
I have installed Pydroid3. On the "Pip" menu option I look for weconnect and it finds it. If I install it, it installs the libraries needed and then weconnect. There is one of these libraries (Pillow) that gives an error when installing it, but somehow if I try again it is finally installed. Not sure if this is relevant.

The error comes when I try to run "weconnect" commands. I.e:

_weconnect-mqtt --help_

I get the following response:

**This copy of software is corrupted and is not safe to use.
To fix this error, uninstall your copy and do a clean install of this app from an official source (e.g. Google Play).**

Any help would be really appreciated.

Many thanks.
BR

**Comment by essjmcb** [comment]
Hi again,

I have changed the python Android apk and now I got close to get something. This is the answer I get from the server:

/weconnect-cli --username <email> --password <pwd>           <
_CRITICAL:weconnect-cli:There was a problem when communicating with WeConnect. If this problem persists please open a bug report: No credentials form found_

I have found a Home Assistant thread from January, where this same error message was shown.

https://community.home-assistant.io/t/add-support-for-vw-id-family/279516/28

Seems that it was fixed straight away. Is it that I am running an old version? This is what I am running:
/weconnect-cli --version                                                      <
weconnect-cli 0.3.0

If it is not up to date, where can I find the latest version?

Thanks!

**Reply by tillsteinbach** [reply]
First question would be: what is your python version?

**Reply by essjmcb** [reply]
Hi,
Thanks for your answer.
It is Python 3.6.6.

**Reply by tillsteinbach** [reply]
You need to have at least python 3.7
Then you should be able to install the latest version

**Reply by essjmcb** [reply]
Many thanks.
Do you have any Python apk suggestion for Android?
- QPython 3L is 3.6.6, so I have the problem above.
- Pydroid 3, is Python 3.9.7, but I get this error:
_This copy of software is corrupted and is not safe to use.
To fix this error, uninstall your copy and do a clean install of this app from an official source (e.g. Google Play)._

**Reply by essjmcb** [reply]
Problem solved on Pydroid 3. There were some env variables missing.
Thanks for your support!
